src/extension/background-script/actions/nostr/delegateOrPrompt.ts

```diff
@@ -0,0 +1,85 @@
+import utils from "~/common/lib/utils";
+import state from "~/extension/background-script/state";
+import { Nip26DelegateConditions } from "~/extension/ln/nostr/types";
+import { MessageDelegateGet, PermissionMethodNostr } from "~/types";
+
+import { addPermissionFor, hasPermissionFor } from "./helpers";
+
+const generateDelegationResponse = async (
+  delegateePubkey: string,
+  conditions: Nip26DelegateConditions
+) => {
+  const publicKey = state.getState().getNostr().getPublicKey();
+  const { cond, sig } = await state
+    .getState()
+    .getNostr()
+    .delegate(delegateePubkey, conditions);
+  const response = {
+    from: publicKey,
```

Comment:
are from and to needed? shouldn't the caller already know these?

Reply:
this compiles with the Delegation interface from `nostr-tools`. I do think it comes in handy to have the same request/response.

I just amended NIP-46 to return this instead of just the sig, for this [reason](https://twitter.com/ArcadeCityMayor/status/1628581196350337025?s=20).

https://github.com/nostr-protocol/nips/blob/master/46.md#optional